apps/web/src/api/v2/util.ts

```diff
@@ -83,6 +101,14 @@ export const createBuild = async (params: {
       complete: false,
     });
 
+    const pullRequest = params.prNumber
+      ? await getOrCreatePullRequest({
+          projectId: params.project.id,
+          number: params.prNumber,
+          trx,
+        })
+      : null;
```

Comment:
Please do it outside transaction. There is no need to do this in the transaction.

Please do it in a lock to ensure there is no concurrency problem while creating the PR.

Reply:
Since I test pull request existence in `getOrCreatePullRequest` function, I think that a transaction would have the same effect than a lock. Don't you agree ?